### PR TITLE
Chart: Fix error when scale breaks are activated (T1222629)

### PR DESCRIPTION
**Pull request by jdvictoria**

### CODE REVIEW

**Comment by jdvictoria** [comment]
[Chart - Error occurs on when scale breaks are activated[T1222629]](https://trello.com/c/3gYW16mv/8641-chart-error-occurs-on-when-scale-breaks-are-activatedt1222629)

File: packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js [hunk 1]
```diff
@@ -798,6 +798,19 @@ QUnit.test('Translate. Update translator with business range with empty scale br
...
 
+QUnit.test('Translate. Scale breaks. Translator should handle division by zero (T1222629)', function(assert) {
+    const breaks = [{ from: -11.9, to: -10.3, cumulativeWidth: 5 }, { from: -10.3, to: 20.3, cumulativeWidth: 10 }];
+    const translator = createTranslatorWithScaleBreaks.call(this, {});
+
+    translator.updateBusinessRange({ min: -11.9, max: 20.3, breaks: breaks });
+
+    const denominator = translator._canvasOptions.rangeMaxVisible - translator._canvasOptions.rangeMinVisible - translator._breaks[translator._breaks.length - 1].length;
+
+    if(denominator === 0) {
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js:809]: to get ratioOfCanvasRange() you can use public translater's API - translator.ratioOfCanvasRange();

File: packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js [hunk 2]
```diff
@@ -798,6 +798,19 @@ QUnit.test('Translate. Update translator with business range with empty scale br
     assert.strictEqual(translator.translate(450), 1000);
 });
 
+QUnit.test('Translate. Scale breaks. Translator should handle division by zero (T1222629)', function(assert) {
+    const breaks = [{ from: -11.9, to: -10.3, cumulativeWidth: 5 }, { from: -10.3, to: 20.3, cumulativeWidth: 10 }];
+    const translator = createTranslatorWithScaleBreaks.call(this, {});
+
+    translator.updateBusinessRange({ min: -11.9, max: 20.3, breaks: breaks });
+
+    const denominator = translator._canvasOptions.rangeMaxVisible - translator._canvasOptions.rangeMinVisible - translator._breaks[translator._breaks.length - 1].length;
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js]: i believe we can just create translator (providing required parametrs, such as canvas and range ) , call updateBusinessRange and after that check translator.ratioOfCanvasRange(), why do we need such calculations in the test? and also the condition checking for zero seems a bit confusing )

File: packages/devextreme/js/viz/translators/translator2d.js [hunk 3]
```diff
@@ -283,8 +283,10 @@ _Translator2d.prototype = {
         canvasOptions.ratioOfCanvasRange = canvasOptions.canvasLength / (canvasOptions.rangeMaxVisible - canvasOptions.rangeMinVisible);
 
         if(breaks !== undefined) {
-            canvasOptions.ratioOfCanvasRange = (canvasOptions.canvasLength - breaks[breaks.length - 1].cumulativeWidth) /
-                (canvasOptions.rangeMaxVisible - canvasOptions.rangeMinVisible - breaks[breaks.length - 1].length);
+            const denominator = canvasOptions.rangeMaxVisible - canvasOptions.rangeMinVisible - breaks[breaks.length - 1].length;
+            if(denominator !== 0) {
```
Comment (AlexanderMoiseev) [inline packages/devextreme/js/viz/translators/translator2d.js]: lets provide more meaningful name for this constant

File: packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js [hunk 4]
```diff
@@ -798,6 +798,19 @@ QUnit.test('Translate. Update translator with business range with empty scale br
...
+
+    if(denominator === 0) {
+        assert.notEqual(translator._canvasOptions.ratioOfCanvasRange, Infinity);
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js]: to get ratioOfCanvasRange() you can use public translater's API - translator.ratioOfCanvasRange();

File: packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js [hunk 5]
```diff
@@ -798,6 +798,15 @@ QUnit.test('Translate. Update translator with business range with empty scale br
     assert.strictEqual(translator.translate(450), 1000);
 });
 
+QUnit.test('Translate. Scale breaks. Translator should handle division by zero (T1222629)', function(assert) {
+    const breaks = [{ from: -11.9, to: -10.3, cumulativeWidth: 5 }, { from: -10.3, to: 20.3, cumulativeWidth: 10 }];
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js:802]: [minor] cant we pass breaks as an arg to createTranslatorWithScaleBreaks? like this const translator = createTranslatorWithScaleBreaks.call(this, { min: 200, max: 700, breaks: [] });

File: packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js [hunk 6]
```diff
@@ -798,6 +798,15 @@ QUnit.test('Translate. Update translator with business range with empty scale br
     assert.strictEqual(translator.translate(450), 1000);
 });
 
+QUnit.test('Translate. Scale breaks. Translator should handle division by zero (T1222629)', function(assert) {
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js]: lets at least name it something like 'Translator should calculate ratioOfCanvasRange correctly'

**Review by AlexanderMoiseev — DISMISSED** [review]

File: packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js [hunk 7]
```diff
@@ -798,6 +798,15 @@ QUnit.test('Translate. Update translator with business range with empty scale br
     assert.strictEqual(translator.translate(450), 1000);
 });
 
+QUnit.test('Translate. Scale breaks. Translator should calculate ratioOfCanvasRange correctly (T1222629)', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.viz.core/translator2D.tests.js]:
```suggestion
QUnit.test('Translate. Scale breaks. Translator should calculate ratioOfCanvasRange correctly when visible range length equals to 0 (T1222629)', function(assert) {
```

**Review by ksercs — APPROVED** [review]
Didn't test, gonna trust u guys

**Review by AlexanderMoiseev — APPROVED** [review]
